### DISCUSSION TITLE
SEB 3.4 on ipados 17.x crashed from Inspera test into blank ipad

### DISCUSSION THREAD

**Discussion by oli-freyr** [SafeExamBrowser/seb-mac#445] [General]
Hi

I'm one of the staff members of the exams office at the University of Iceland. A student came to me with this problem.
I hope that I'm posting this in the right place.

According to the student, he was in the middle of an exam curated by Inspera when suddenly the screen went blank and only the SEB icon is visible.
I include a picture of the ipad in this state with the file Screen1-blank.jpg
When I tap the app icon for SEB, there is a message about screen proctoring, the file Screen2-message.jpg shows that state.
Screen1-blank.jpg
![Sceen1-blank](https://github.com/user-attachments/assets/5b5b99a0-5681-429a-a7c3-5ef273036cf2)
![Screen2-message](https://github.com/user-attachments/assets/8433edb8-d5aa-4aa2-a048-09d5d5fff20b)

As the ipad is now, there is no other app available but most of the gestures are possible:
* Swipe up short distance to close SEB.
* Swipe up long distance to see list of open apps, only SEB is there. 
* a few others.

Notable absense is Control Center when I swipe down the right side of the screen.
I can provide a video of my interaction with the ipad if needed.

I tried to shut down the ipad by holding both power button and volume button together and swipe the shutdown slider.
I tried to force restart the ipad by pressing volume up, down and power buttons.

Both times, the ipad reboots and comes back into this state.

My questions are therefore: Is this a known state? Is this a new functionality of the kiosk lockdown? Is there an easy way out?

This looks like everything has been deleted off the ipad. That is very hard to believe.

I am now in the first few minutes of diagnosing this problem and would like to send you any log files. Where can I find them?
Can anyone recommend any applications for my mac to extract a backup before I perform a factory reset / reinstallation?

Any help is greatly appreciated and of course if there is a new bug exposed, I will assist with any information you request.

Best regards,
Óli

**Comment by danschlet** [comment]
It seems the iPad is stuck in the new multi-app assessment (AAC) mode. Could it be that the exam didn't had an SEB administrator password set, so the student was able to open the in-app settings in SEB?

Also, is this an MDM-managed iPad (usually belonging to the university/school)? That would explain why it it stuck in this mode after a reboot.

**Reply by oli-freyr** [reply]
Hi danschlet
Thank you for responding. There is a *.seb config file downloaded from Inspera which provides the quit password. When SEB crashed it seems to not have that password available.
If I tap on the Quit button in the lower right corner, nothing happens. No prompt for a password.

The ipad belongs to the student. It is not managed by the University.

**Reply by danschlet** [reply]
Ok. This is weird. Are you sure the iPad returns to this state even after a hard reboot/force reset? That was never the case with the AAC assessment mode. Only if SEB is set as an application which is allowed to start the ASAM (Autonomous Single App Mode), then after a restart the assessment app is restarted automatically. This is a feature which usually requires MDM (maybe it's also possible to do using Apple Configurator, I'm not sure about that).

If SEB is really restarted and stuck in this mode, that would be a serious iOS/iPadOS bug which we urgently need to report to Apple. 
Which iPadOS version is the iPad using exactly? 17.7?

Also, when SEB is restarted, it should still ask "Confirm App Self-Lock – SEB wants to start Assessment Mode. You will be unable to use other apps on iPad until SEB ends Assessment Mode. Do you want to allow this?". This is a system dialog and if you tap Cancel, the iPad should not be locked.

Also it doesn't make sense that the iPad shows the screen proctoring session warning, as that has to be configured first. It looks like the student tried to change the client settings of his SEB installation prior to his Inspera exam. Or he clicked on one of our demo/testing links, as I had a configuration there once where screen proctoring was falsely enabled. Also the student should use the official App Store release of SEB, not one of the TestFlight Betas.

**Reply by danschlet** [reply]
What happens if you tap "Cancel" in the screen proctoring session warning? The exam session should not start then.

Also, what happens if you tap the SEB quit button (right most bottom icon)? Does it ask for a quit password? What does it show after quitting?

**Reply by oli-freyr** [reply]
Hi danschlet
I performed reboots a few times and every time, the ipad automatically enters the same mode, no questions asked.
The student claims that the app was downloaded from App Store and this is just his private ipad, no MDM. The student is not studying Computer Science or related line of study and shows average computer literacy.
I don't have access to see exactly the ipad version, believe close to the latest update of version 17.

The Cancel button just causes the dialog to return. Endless loop.

Other buttons at the bottom, SEB icon and quit button are not active. Nothing happens when I try to tap.

**Reply by danschlet** [reply]
Ok, this definitely must be an iPadOS bug related to the AAC [Assessment Mode](https://support.apple.com/en-us/101976).

I will report this to Apple today and hope to get some feedback about what can be done to get the iPad out of this.

I would try to connect the iPad to a Mac running Apple Configurator and see if you get some options there to reset the state the iPad is in. You should probably see the exact iPadOS version there as well.

Usually putting iOS devices into DFU mode allows some deeper kind of reset. Also you might be able to update the iPadOS version with Apple Configurator, although it might delete all content or there might be a risk to loose data, even if it doesn't perform a full reset of the device data.

**Reply by oli-freyr** [reply]
Thank you for reporting this. I will experiment with Apple Configurator. Not an expert on that, so I might have questions.

**Reply by oli-freyr** [reply]
For your amusement, here is a video of restarting the ipad from SEB, back to SEB:

https://github.com/user-attachments/assets/3979d7a1-77a6-4f9d-b259-9387b934f5bd

**Reply by oli-freyr** [reply]
I now have detailed information about the hardware and software of the ipad.
[SEB Ipad.txt](https://github.com/user-attachments/files/17311467/SEB.Ipad.txt)

**Comment by danschlet** [comment]
To send log files, if you can go into the left side menu (swipe right from left screen edge or tap SEB icon on the bottom left):

1. - Tap "About SEB"
2. - Scroll completely down 
3. - Tap "Send Logs to SEB Developers"
4. - E-mail the logs to yourself and attach it here

If you're in the Initial Configuration Assistant view, tap the SEB icon there and continue with step 2 above.

**Reply by oli-freyr** [reply]
Unfortunately, the swipe gesture doesn't work and the SEB icon doesn't accept tap.
To test this I have pressed cancel on the Proctoring dialog box and the "Options to start an exam or configure SEB for your institution" dialog box is active.
Screen image:
![Screen4-config](https://github.com/user-attachments/assets/47512547-6c5f-45e9-a963-d991f5aaf0a2)

Gestures:

https://github.com/user-attachments/assets/b13b8735-a1f6-4838-bcd1-aaf7d9056516

**Reply by danschlet** [reply]
No, this is expect behavior, when such an modal view is displayed. You can tap the upper SEB icon partially covered by the title of the view („Options to start an exam or configure SafeExamBrowser for your institution“). Then the About SEB view is displayed where you can send the log files.

**Reply by danschlet** [reply]
Try this (but please try to send the log files first):

1. In the text field below „Enter the URL of an institution which supports SEB“ enter safeexambrowser.org
2. SEB should be reconfigured and display a list of exams. Check if the iPad got unlocked (other icons visible on the Home Screen)
3. If not, tap the first link „SEB Demo Exam“. Please film what happens
4. The question about starting assessment mode should be displayed, confirm that
5. If SEB is started into an exam, tap the quit button and enter the quit password „quit“
6. Maybe now the iPad is back to a normal mode?

**Reply by oli-freyr** [reply]
I tap the SEB logo and scroll to the bottom.
Tap the link to "Send Logs to SEB Developers" but get error dialog box
"Can't Send Log File"
"This Device isn't configured for sending email"

Probably because the student hasn't set up an email account in Settings.

**Reply by oli-freyr** [reply]
I sent you an email. I hope it's ok.

**Reply by oli-freyr** [reply]
Hi danschlet
I tried entering safeexambrowser.org as the URL and successfully connected to the test list.
When I entered the SEB Demo Exam and was presented with a blank page.
This time, the SEB logo in the bottom left corner and the quit button in the bottom right corner are active.
I tapped the quit button and entered the password.
This was not enough.
I exited SEB and was back in the limited environment.
However, when I started SEB again, it presented with a new dialog box "No Kiosk Mode Available":
![Screen7-no-kiosk](https://github.com/user-attachments/assets/301ea75a-fb87-4fc7-a5a1-611c8e191ec5)

When I tap Retry, the dialog box disappeared and I was back in the list of tests in SEB.
When I close the app, I'm back in the normal ipad environment.

Thank you very much, I appreciate the help tremendously.
Unfortunately, I was unable to find the log file specific to SEB. I have returned the device to it's owner.

Best regards,
Óli
